### PR TITLE
Avoid code which looks at local files when etcd is configured

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Avoid code which looks at local files when etcd is configured

## Motivation and Context
This situation happens only in gateway nas which supports
etcd based `config.json` to support all FS mode features.

The issue was we would try to migrate something which doesn't
exist when etcd is configured which leads to inconsistent
server configs in memory.

This PR fixes this situation by properly loading config after
initialization, avoiding backend disk config migration to be
done only if etcd is not configured.

## Regression
No, continuation of the fix in #7134

## How Has This Been Tested?
Involving the user directly with a custom image provided with the fix `y4m4/minio:jody`, requires one to use helm chart based `gateway nas` deployment with etcd, identity, OPA configured. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
ping @poornas @vadmeste

**Review by poornas — DISMISSED** [review]

File: cmd/config.go [hunk 1]
```diff
@@ -172,6 +174,10 @@ func initConfig(objAPI ObjectLayer) error {
 				return err
 			}
 		}
+
+		// Watch config for changes and reloads them i
+		go watchConfig(objAPI, configFile, loadConfig)
```
Comment (vadmeste) [inline cmd/config.go]: Shouldn't we also watch for config file in case of distributed XL ?

**Review by vadmeste — COMMENTED** [review]
Question

File: cmd/config.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/config.go]: watchConfig is only etcd call, it should be renamed.

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Pull request merged**